x-pack/plugins/observability_solution/entity_manager/server/routes/entities/update.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from 'zod';
+import { RequestHandlerContext } from '@kbn/core/server';
+import {
+  createEntityDefinitionQuerySchema,
+  CreateEntityDefinitionQuery,
+  entityDefinitionUpdateSchema,
+  EntityDefinitionUpdate,
+} from '@kbn/entities-schema';
+import { SetupRouteOptions } from '../types';
+import { EntitySecurityException } from '../../lib/entities/errors/entity_security_exception';
+import { InvalidTransformError } from '../../lib/entities/errors/invalid_transform_error';
+import { startTransform } from '../../lib/entities/start_transform';
+import {
+  installationInProgress,
+  reinstallEntityDefinition,
+} from '../../lib/entities/install_entity_definition';
+import { findEntityDefinitionById } from '../../lib/entities/find_entity_definition';
+
+/**
+ * @openapi
+ * /internal/entities/definition:
+ *   put:
+ *     description: Update an entity definition.
+ *     tags:
+ *       - definitions
+ *     parameters:
+ *       - in: query
+ *         name: installOnly
+ *         description: If true, the definition transforms will not be started
+ *         required: false
+ *         schema:
+ *           type: boolean
+ *           default: false
+ *     requestBody:
+ *       description: The definition properties to update
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/schemas/entityDefinitionUpdateSchema'
+ *     responses:
+ *       200:
+ *         description: Success
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/entityDefinitionSchema'
+ *       400:
+ *         description: The entity definition cannot be installed; see the error for more details
+ *       404:
+ *         description: The entity definition does not exist
+ *       403:
+ *         description: User is not allowed to update the entity definition
+ *       409:
+ *         description: The entity definition is being updated by another request
+ */
+export function updateEntityDefinitionRoute<T extends RequestHandlerContext>({
+  router,
+  server,
+}: SetupRouteOptions<T>) {
+  router.put<{ id: string }, CreateEntityDefinitionQuery, EntityDefinitionUpdate>(
```

Comment:
Since it is a partial (by design) change, we should use PATCH instead of PUT, based on [the API guidelines](https://github.com/elastic/observability-dev/blob/main/docs/observability-experience/low-level-api-guidelines.md#update)